Improve docs (#104)
gardener/test-infra: docs/GetStarted.md

- ginkgo test suite at $GOPATH/src/github.com/onsi/ginkgo
- Go project setup script
  - automatically setup test repository at the provided gopath and cd's into it
  - RUN ```/tm/setup github.com/org repo``` e.g. ``` /tm/setup github.com/gardener test-infra ```

### Test

> A TestDefinition can be easily tested with the Test Machinery wiht the following steps:

Reviewer: ```suggestion
A TestDefinition can be easily tested with the Testmachinery with the following steps:
```

- Upload your Test and TestDefinition e.g. `my-test` to your repo e.g. `github.com/my-repo/it.git`
- Copy the example Testrun for a single test from `/examples/single-testrun.yaml`
- Add your test repository to the `locationSets` of the copied Testrun:
  ```yaml
  spec:
    locationSets:
    - name: default
      default: true
      locations:
      - type: git
        repo: github.com/my-repo/it.git
        revision: master
  ```

> - Add your testdefinition name to `testflow[0]definition.name` of the copied testrun

Reviewer: ```suggestion
- Add your TestDefinition name to `testflow[0]definition.name` of the copied Testrun
```

  ```yaml
  spec:
    testflow:
    - name: test
      definition:
        name: my-test
  ```

> - Add your dependent kubeconfigs to `kubeconfigs` of the testrun.

Reviewer: ```suggestion
- Add your dependent kubeconfigs to `kubeconfigs` of the Testrun.
```

>   For exmaple if you just want to test a already existing shoot. Just base64 encode the kubeconfig `cat $KUBECONFIG | base64 -w0` and copy the string to `kubeconfigs.shoot`.

Reviewer: ```suggestion
  For example if you just want to test an already existing shoot, then just base64 encode the kubeconfig `cat $KUBECONFIG | base64 -w0` and copy the string to `kubeconfigs.shoot`.
```

  The same procedure applies for other kubeconfigs (but all kubeconfigs are optional).

> - Run the testrun with `kubectl create -f single-testrun.yaml` (Note: your current kubecontext need to point to the cluster where the testmachinery is installed).

Reviewer: ```suggestion
- Run the Testrun with `kubectl create -f single-testrun.yaml` (Note: your current kubecontext need to point to the cluster where the Testmachinery is installed).
```
